### ISSUE TITLE
[no-unused-var] False positive for imported interfaces used in declare global

### ISSUE THREAD

**Issue by flopes89** [typescript-eslint/typescript-eslint#1302]
**Repro**

```JSON
{
  "plugins": ["react", "@typescript-eslint"],
  "extends": ["eslint:recommended", "plugin:react/recommended", "plugin:@typescript-eslint/recommended"],
  "rules": {
    "no-unused-vars": "warn",
  }
}
```

```TS
// ../server/servicesTypes.ts

export interface Token {
	principal_id: string;
}
```

```TS
import { Token } from "../server/servicesTypes";

declare global {
	namespace Express {
		export interface Request {
			token: Token;
		}
	}
}

```

**Expected Result**

File does not yield any warnings.

**Actual Result**

ESLint warning on the import line:

> ESLint: 'Token' is defined but never used. (no-unused-vars)

**Additional Info**

When I remove the `declare global` and instead `export namespace` is the top-most declaration, the warning goes away.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.6.0` |
| `@typescript-eslint/parser`        | `2.6.0` |
| `TypeScript`                       | `3.6.2` |
| `ESLint`                           | `6..6.0` |
| `node`                             | `10.15.3` |
| `npm`                              | `6.4.1` |

**Comment by bradzacher** [comment]
please use our rule `@typescript-eslint/no-unused-vars-experimental` instead

**Comment by ma-zal** [comment]
@bradzacher Can you provide details, how to use it? I was not able to found the documentation.

> please use our rule `@typescript-eslint/no-unused-vars-experimental` instead

**Comment by armano2** [comment]
@ma-zal https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars-experimental.md
